feat(lgr): add lgr functionality to frontend

GauChoob opened the pull request.

GauChoob commented:
Adds an LGR tab on the left, allowing you to upload and view lgrs

metsavir commented on README.md:
```diff
@@ -115,6 +115,7 @@ The config is set up to use a test database. Bear in mind the following:
 - Test database is a copy of live from end june 2020
 - time, battle and allfinished tables only have a subset of live (last year or two) to keep size down
 - Passwords, emails and private comments have been stripped
+- You can rename .env.test -> VITE_API_URL=http://localhost:3003/ if you are developing the backend at the same time.
```
`yarn start:dev` uses local backend. It gets env from `.env`.

GauChoob replied:
Kind of my fault for not checking closely. Thanks!

sunehs commented on src/components/Dropzone.jsx:
```diff
@@ -5,8 +5,9 @@ import config from 'config';
 const types = {
-  '.rec': { '*/*': ['.rec'] },
-  '.dat': { '*/*': ['.dat'] },
+  '.rec': { 'application/octet-stream': ['.rec'] },
+  '.dat': { 'application/octet-stream': ['.dat'] },
+  '.lgr': { 'application/octet-stream': ['.lgr'] },
```
for non standard file types like these there's no guarantee they're application/octet-stream, so we need to use wildcard

GauChoob replied:
Thanks for the review!

I had changed this to fix a warning that is thrown on Chrome and Firefox:

Skipped "*/*" because it is not a valid MIME type. Check https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types for a list of valid MIME types.
![image](https://github.com/user-attachments/assets/af73c34e-5a01-4814-96fc-1a6c2e6aa56a)

Let me know if you still want me to revert it!

sunehs replied:
in the past I've had to add all kinds of wacky mime types everytime some couldn't upload their rec file, so I think we have to keep the wildcard to be safe

sunehs commented on src/features/LGRUpload/index.jsx:
```diff
@@ -0,0 +1,360 @@
+      const res = await EditLGR(lgrToEdit.LGRName, formData);
+      if (res.data && !res.data.error) {
+        if (lgrData.filename !== lgrToEdit.LGRName) {
+          window.location.href = `${window.location.origin}/l/${lgrData.filename}`;
```
for navigation use useNavigate from @tanstack/react-router

```
import { useNavigate } from '@tanstack/react-router';
const navigate = useNavigate();
navigate({ to: `/l/${lgrData.filename}` });
```

sunehs commented on src/pages/lgr/index.jsx:
```diff
@@ -0,0 +1,175 @@
+  const deleteLGR = async () => {
+    const res = await DeleteLGR(lgr.LGRName);
+    if (res.ok) {
+      window.location.href = `${window.location.origin}/lgrs`;
```
useNavigate

sunehs reviewed (commented):
looks pretty solid, just a couple minor things, nice work for a first contribution getting all these features included :p

GauChoob commented:
I fixed the 3 things you highlighted. I also added a small feature where mods can change ownership of lgrs. I also made the frontend source the preview images directly from the S3 database, hopefully cached, but I can't test this since I don't actually have access to S3 and am just testing things locally.
